packages/controller/lib/setup.js

```diff
@@ -600,82 +600,109 @@ async function processCommand(command, args, params, callback) {
+                        const repo = new Repo({ objects, states });
+
+                        try {
+                            await repo.rename('default', 'stable', 'http://download.iobroker.net/sources-dist.json');
+                            await repo.rename(
+                                'latest',
+                                'beta',
+                                'http://download.iobroker.net/sources-dist-latest.json'
+                            );
+                        } catch (err) {
+                            console.warn(`Cannot rename: ${err.message}`);
+                        }
+
+                        // there has been a bug that user can upload js-controller
+                        try {
+                            await objects.delObjectAsync('system.adapter.js-controller');
+                        } catch {
+                            // ignore
+                        }
+
+                        try {
+                            const configFile = tools.getConfigFileName();
+
+                            const configOrig = fs.readJSONSync(configFile);
+                            const config = deepClone(configOrig);
+
+                            config.objects.options = config.objects.options || {
+                                auth_pass: null,
+                                retry_max_delay: 5000
+                            };
+                            if (
+                                config.objects.options.retry_max_delay === 15000 ||
+                                !config.objects.options.retry_max_delay
+                            ) {
+                                config.objects.options.retry_max_delay = 5000;
+                            }
+                            config.states.options = config.states.options || {
+                                auth_pass: null,
+                                retry_max_delay: 5000
+                            };
+                            if (
+                                config.states.options.retry_max_delay === 15000 ||
+                                !config.states.options.retry_max_delay
+                            ) {
+                                config.states.options.retry_max_delay = 5000;
                             }
 
-                            // there has been a bug that user can upload js-controller
-                            try {
-                                await objects.delObjectAsync('system.adapter.js-controller');
-                            } catch {
-                                // ignore
+                            let migrated = false;
+                            // We migrate file to jsonl
+                            if (config.states.type === 'file') {
+                                config.states.type = 'jsonl';
+                                console.log('States DB type migrated from "file" to "jsonl"');
+                                migrated = true;
```

Comment:
maybe remember which db was cchanged to add to the notification?

Reply:
yes, next steps